sharding/nodesCoordinator/indexHashedNodesCoordinator.go

```diff
@@ -277,6 +276,38 @@ func (ihnc *indexHashedNodesCoordinator) getNodesConfig(epoch uint32) (*epochNod
 	return nodesConfigEpoch, ok
 }
 
+func (ihnc *indexHashedNodesCoordinator) getNodesConfigFromStorer(epoch uint32) ([]byte, error) {
+	var err error
+
+	mostRecentEpoch := ihnc.getMostRecentLoadedEpoch()
```

Comment:
This should be returning the most recent epoch that would have (according to the config) the requested epoch config (parameter)

e.g 
a. we are in epoch 10
b. config is to keep latest 4 epochs in memory
c. request is to get the nodes config for epoch 2

according to (b.) epoch 2 config should be available (in memory and also stored on disk) in epochs 2, 3, 4, 5.
the most recent epoch from those would be epoch 5, (you wouldn't be able to find the config stored in persisters for epochs 6, 7, 8, 9, 10).

So in this case you should try to get it from epoch 5 persister `ihnc.bootStorer.GetFromEpoch(ncInternalkey, 5)`
maybe also do a retrial if not found with previous epoch `ihnc.bootStorer.GetFromEpoch(ncInternalkey, 4)`